Automattic/wp-calypso
# A4A: Integrate Users REST endpoint to the team member list.

This PR wires the User's endpoint to the team members' page.

<img width="1727" alt="Screenshot 2024-08-27 at 7 13 09 PM" src="https://github.com/user-attachments/assets/c823168c-cb01-4921-a730-06e15aa01b70">


Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/943


**Please ensure that this is only merged if D159377-code has been merged.**

## Proposed Changes

* Added a `useTeamMember` hook to handle both the fetch request for the users and the user invites endpoint. Consolidate the results into one member list, which is then outputted to the members table.
* Additionally, update the User invite hook to include additional fields (avatar icon and email).

## Why are these changes being made?

*

## Testing Instructions


* Apply the D159377-code to your sandbox and point `public-api.wordpress.com` to your sandbox.
* Use the A4A live link and go to the `/team` page.
* Confirm that the UI fetches data from the Users and Users Invite endpoints. 
* Add some users or create some invites.
* Confirm that it appears in the table.

## Pre-merge Checklist
